### DISCUSSION TITLE
Support store metadata with rpc service not with db or KV directly

### DISCUSSION THREAD

**Discussion by rongfengliang** [juicedata/juicefs#1300] [General]
**What would you like to be added**:

currently `juicefs`  metadata storage use DB or KV  storage. this is not good  for security or  management  in enterprise 

**Why is this needed**:

can we use RPC service or other  for  good  management  I know with RPC or others may be difficult  (transaction is one most important thing )

**Comment by davies** [comment]
If we create a RPC service for JuiceFS, then we should provide the way and tools for management, those will be similar to that provided by existing database (Redis and other SQL databases). So, I think reuse the existing tools to manage DB or KV is better than re-build them.

JuiceFS serves as a bottom infrastructure tool, we can build some tool to manage it, to avoid expose the database directly to end users.

Adding a RPC service will increase latency to meta operation, which is critical to JuiceFS, we'd like to avoid that as much as possible.

**Reply by rongfengliang** [reply]
@davies  Yes, I agree,  Plan to add  feature like  juicefs pro  so  we can use token to manage backend metadata storage
